superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -299,19 +317,42 @@ const displayField = ({
   getValidation,
   validationErrors,
   db,
+  setPublicSheets,
 }: FieldPropTypes) => (
-  <ValidatedInput
-    id="database_name"
-    name="database_name"
-    required
-    value={db?.database_name}
-    validationMethods={{ onBlur: getValidation }}
-    errorMessage={validationErrors?.database_name}
-    placeholder=""
-    label="Display Name"
-    onChange={changeMethods.onChange}
-    helpText={t('Pick a nickname for this database to display as in Superset.')}
-  />
+  <>
+    <ValidatedInput
+      id="database_name"
+      name="database_name"
+      required
+      value={db?.database_name}
+      validationMethods={{ onBlur: getValidation }}
+      errorMessage={validationErrors?.database_name}
+      placeholder=""
+      label="Display Name"
+      onChange={changeMethods.onChange}
+      helpText={t(
+        'Pick a nickname for this database to display as in Superset.',
+      )}
+    />
+
+    {db?.engine === 'gsheets' && (
+      <>
+        <FormLabel required>{t('Type of Google Sheets Allowed')}</FormLabel>
+        <Select
+          style={{ width: '100%' }}
+          onChange={(value: string) => setPublicSheets(value)}
+          defaultValue="true"
+        >
+          <Select.Option value="true" key={1}>
+            Publicly shared sheets only
+          </Select.Option>
+          <Select.Option value="false" key={2}>
+            Public and privately shared sheets
+          </Select.Option>
```

Comment:
If you set the values to booleans you don't need to convert them from string:

```suggestion
          <Select.Option value={true} key={1}>
            Publicly shared sheets only
          </Select.Option>
          <Select.Option value={false} key={2}>
            Public and privately shared sheets
          </Select.Option>
```

Reply:
Yeah I tried this but the antd component only allows string or number:
https://3x.ant.design/components/select/

That's why I did the string conversion. Can we change that locally?

Reply:
Ah, good point. I would convert it as soon as possible then, passing around a string that should really be a boolean is prone to bugs:

```js
<Select
          style={{ width: '100%' }}
          onChange={(value: string) => setPublic(stringToBoolean(value))}
```

Where:

```typescript
const stringToBoolean = (value: string): boolean => value === 'true';
```

Reply:
good idea, I will make that change right now.